app/app.go

```diff
@@ -526,6 +529,23 @@ func (app *App) Info(req *abci.RequestInfo) (*abci.ResponseInfo, error) {
 	return res, nil
 }
 
+// FinalizeBlock implements the abci interface. It overrides baseapp's FinalizeBlock method, essentially becoming a decorator
+// in order to add transaction pruning logic after normal finalize block processing.
+func (app *App) FinalizeBlock(req *abci.RequestFinalizeBlock) (*abci.ResponseFinalizeBlock, error) {
+	// Call the normal BaseApp FinalizeBlock first
+	res, err := app.BaseApp.FinalizeBlock(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// go through all the transactions that are getting executed and prune the tx tracker
+	for _, tx := range req.Txs {
+		app.txValidationCache.RemoveTransaction(tx)
+	}
```

Comment:
[non blocking]
would it make sense to run these 2 parallel? just in case removing the transactions from the cache is heavy. Otherwise it's fine

Reply:
why 20? worst case is 200